main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -384,7 +384,44 @@ object Verifier {
           check(expected = MonoType.Region)(actual = t2, loc)
           check(expected = MonoType.Unit)(actual = tpe, loc)
 
-        case _ => tpe // TODO: VERIFIER: Add rest
+        case AtomicOp.GetField(field) =>
+          val List(t) = ts
+          checkJavaSubtype(t, field.getDeclaringClass, loc)
+          checkJavaSubtype(tpe, field.getType, loc)
+
+        case AtomicOp.GetStaticField(field) =>
+          checkJavaSubtype(tpe, field.getType, loc)
+
+        case AtomicOp.PutField(field) =>
+          val List(t1, t2) = ts
+          checkJavaSubtype(t1, field.getDeclaringClass, loc)
+          checkJavaSubtype(t2, field.getType, loc)
+          check(expected = MonoType.Unit)(actual = tpe, loc)
+
+        case AtomicOp.PutStaticField(field) =>
+          val List(t) = ts
+          checkJavaSubtype(t, field.getType, loc)
+          check(expected = MonoType.Unit)(actual = tpe, loc)
+
+        case AtomicOp.InstanceOf(_) =>
+          val List(t) = ts
+          checkJavaSubtype(t, new Object().getClass, loc) // must not be primitive type
```

Comment:
Is this a requirement in typechecking? if so just close this comment, just wanted to make sure you've checked.

Reply:
Yea, in Java, `e instanceof ...` requires that `e` is a reference (from my testing...)